pandas/core/groupby/groupby.py

```diff
@@ -898,6 +898,11 @@ def _python_agg_general(self, func, *args, **kwargs):
         # iterate through "columns" ex exclusions to populate output dict
         output = {}
         for name, obj in self._iterate_slices():
+            if self.grouper.ngroups == 0:
+                # agg_series below assumes ngroups > 0
+                #  3 test cases get here
```

Comment:
Nit: somewhat on the fence regarding the necessity and usefulness of this comment

Reply:
fair enough.  its for my own reference in trying to move these checks earlier

Reply:
Similar comment on another PR but would ideally not special case things like this in groupby functions. Is there an actual issue in allowing this to follow the same code paths as when ngroups is non-zero?

Reply:
@WillAyd that goal is the reason for the "3 test cases" comment (which I'll remove).  I want to identify the cases that get here and make this check unnecessary.  But its much clearer to do this check and not end up with `None` on L914 and not be clear on how that happened.